### PR TITLE
Replace ZOWE_EDITOR with ZOWE_OPT_EDITOR for consistency

### PR DESCRIPTION
**Pull request by t1m0thyj**
**What It Does**
* Replaces the `ZOWE_EDITOR` environment variable with `ZOWE_OPT_EDITOR` and `--editor` option on commands
* Removes check for `ZOWE_EDITOR` environment variable in `ProcessUtils.openInEditor`

**How to Test**
Run the following commands with the `--editor` option:
* `zowe config edit`
* `zowe config auto-init --edit`
* `zowe config init --edit`

**Review Checklist**
I certify that I have:
- [ ] tested my changes
- [ ] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

File: packages/imperative/src/imperative/src/config/cmd/edit/edit.definition.ts [hunk 1]
```diff
@@ -24,8 +24,7 @@ export const editDefinition: ICommandDefinition = {
     summary: "edit config files",
     description: `Edit an existing config file in your system's default text editor.\n\n` +
         `In a graphical environment, the application associated with JSON files will be launched. ` +
-        `In a command-line environment, vi will be launched. To override the command-line editor used, specify it in ` +
-        `the ${ImperativeConfig.instance.loadedConfig.envVariablePrefix}_EDITOR environment variable.`,
```
Comment (ATorrise) [inline packages/imperative/src/imperative/src/config/cmd/edit/edit.definition.ts:28]: lint error - ImperativeConfig

**Review by ATorrise — CHANGES_REQUESTED** [review]

File: packages/imperative/src/imperative/src/config/cmd/edit/edit.definition.ts (hunk 1 above)
Comment (gejohnston) [reply packages/imperative/src/imperative/src/config/cmd/edit/edit.definition.ts:28]: 👍Also occurs in packages/imperative/src/utilities/src/ProcessUtils.ts

**Review by gejohnston — CHANGES_REQUESTED** [review]

**Review by gejohnston — APPROVED** [review]

**Review by awharn — APPROVED** [review]

**Review by traeok — APPROVED** [review]
LGTM, thanks @t1m0thyj

**Review by zFernand0 — COMMENTED** [review]
thanks for fixing the changelog

**Pull request merged**